### PR TITLE
Lightning Boost

### PR DESCRIPTION
**Pull request by dannydeezy**
Integrates the Deezy UTXO Boost endpoint if the value of the output falls below 600 sats

### CODE REVIEW

File: src/components/modals/send-modal/index.js [hunk 1]
```diff
@@ -34,7 +36,40 @@ const SendModal = ({ show, handleModal, utxo, onSale }) => {
         }
     }, []);
 
-    async function sendUtxo() {
+    async function sendUtxo(boost = false) {
```
Comment (habibitcoin) [inline src/components/modals/send-modal/index.js:39]: nit: Is this param actually necessary? Would be fairly lightweight to calculate on the fly

File: src/components/modals/send-modal/index.js [hunk 2]
```diff
@@ -34,7 +36,40 @@ const SendModal = ({ show, handleModal, utxo, onSale }) => {
...
+                const signedTxHex = await createAndSignPsbtForBoost({
+                    pubKey: nostrPublicKey,
+                    utxo,
+                    destinationBtcAddress,
+                });
+                const { data } = await axios.post(`https://api${TESTNET ? "-testnet" : ""}.deezy.io/v1/boost`, {
+                    psbt: signedTxHex,
+                    fee_rate: sendFeeRate,
+                });
+                console.log(data);
```
Comment (habibitcoin) [inline src/components/modals/send-modal/index.js:60]: nit: remove logs

File: src/components/modals/send-modal/index.js [hunk 3]
```diff
@@ -74,12 +108,13 @@ const SendModal = ({ show, handleModal, utxo, onSale }) => {
...
 
+    const boostRequired = !!utxo && !!sendFeeRate && outputValue(utxo, sendFeeRate) < MIN_OUTPUT_VALUE;
+
     const submit = async () => {
         setIsSending(true);
-        await sendUtxo().catch((err) => {
+        await sendUtxo(boostRequired).catch((err) => {
             console.error(err);
             alert(err);
-            return false;
```
Comment (habibitcoin) [inline src/components/modals/send-modal/index.js:82]: nit: Why removal of true/false returns?n

File: src/utils/psbt.js [hunk 4]
```diff
@@ -102,6 +108,21 @@ function createPsbt({ utxo, inputAddressInfo, destinationBtcAddress, sendFeeRate
     return { psbt, inputParams };
 }
 
+function createPsbtForBoost({ utxo, inputAddressInfo, destinationBtcAddress }) {
+    const network = TESTNET ? bitcoin.networks.testnet : bitcoin.networks.bitcoin;
+
+    const psbt = new bitcoin.Psbt({ network });
+    const sighashType = 131; // bitcoin.Transaction.SIGHASH_SINGLE | bitcoin.Transaction.SIGHASH_ANYONECANPAY;
```
Comment (habibitcoin) [inline src/utils/psbt.js:115]: nit: maybe a config value?

File: src/utils/psbt.js [hunk 5]
```diff
@@ -102,6 +108,21 @@ function createPsbt({ utxo, inputAddressInfo, destinationBtcAddress, sendFeeRate
     return { psbt, inputParams };
 }
 
+function createPsbtForBoost({ utxo, inputAddressInfo, destinationBtcAddress }) {
+    const network = TESTNET ? bitcoin.networks.testnet : bitcoin.networks.bitcoin;
```
Comment (habibitcoin) [inline src/utils/psbt.js:112]: nit: I think already a config value?

File: src/utils/psbt.js [hunk 6]
```diff
@@ -112,11 +133,21 @@ async function broadcastPsbt(psbt) {
 }
 
 export async function signAndBroadcastUtxo({ pubKey, utxo, destinationBtcAddress, sendFeeRate }) {
-    const inputAddressInfo = await getAddressInfo(pubKey);
+    const inputAddressInfo = getAddressInfo(pubKey);
```
Comment (habibitcoin) [inline src/utils/psbt.js:136]: Why removal of await?

File: src/utils/psbt.js [hunk 7]
```diff
@@ -112,11 +133,21 @@ async function broadcastPsbt(psbt) {
...
     const signed = await getSignedPsbt({ psbt, inputParams, inputAddressInfo, utxo });
     // Finalize the PSBT. Note that the transaction will not be broadcast to the Bitcoin network yet.
     signed.finalizeAllInputs();
     // Send it!
     return broadcastPsbt(signed);
 }
+
+export async function createAndSignPsbtForBoost({ pubKey, utxo, destinationBtcAddress }) {
+    const inputAddressInfo = getAddressInfo(pubKey);
+    const { psbt, inputParams } = createPsbtForBoost({ utxo, inputAddressInfo, destinationBtcAddress });
```
Comment (habibitcoin) [inline src/utils/psbt.js:147]: nit: can merge createPsbtForBoost inside of createAndSignPsbtForBoost()

Think createPsbtForBoost() might be over-abstracting

**Review by habibitcoin — APPROVED** [review]
functional and good to ship but have some nits for follow up:
- metamask
- TIL about webln, should we be implementing this more broadly to support more bitcoin wallets? Is it that easy or am I oversimplifying our ability to support other webln wallets? @dannydeezy

File: src/components/modals/send-modal/index.js (hunk 3 above)
Comment (dannydeezy) [reply src/components/modals/send-modal/index.js:82]: the linting forced me to be consistent, either always return something or never return something. we can put them all back and add ones for the new returns i added

File: src/utils/psbt.js (hunk 6 above)
Comment (dannydeezy) [reply src/utils/psbt.js:136]: its not an async function

**Comment by dannydeezy** [comment]
> functional and good to ship but have some nits for follow up:
> 
> * metamask
> * TIL about webln, should we be implementing this more broadly to support more bitcoin wallets? Is it that easy or am I oversimplifying our ability to support other webln wallets? @dannydeezy

thanks, i made an issue to track the nits
https://github.com/dannydeezy/nosft/issues/149

**Pull request merged**